version_template.py

```diff
@@ -107,14 +102,29 @@ def buildNumber():
 
 
 def currentCommit():
+    import os
     from subprocess import check_output
-    return check_output('git log --pretty=oneline -n 1 -- $(pwd)', shell=True).split()[0]
+
+    return check_output(['git', 'log', '--pretty=oneline', '-n', '1', os.getcwd()]).split()[0]
 
 
 def dirty():
+    """
+    Returns True if the code has uncommitted diffs.
+
+    :return: bool
+    """
     from subprocess import call
-    return 0 != call('(git diff --exit-code '
-                     '&& git diff --cached --exit-code) > /dev/null', shell=True)
+    import os
+
+    with open(os.devnull, 'w')  as FNULL:
+        # return code is 1 if there are still changes relative to the index
+        # otherwise return code is 0
+        retcode1 = call(['git', 'diff', '--exit-code'], stdout=FNULL)
+    # return code is 1 if there are still changes staged for the next commit relative to HEAD
+    # otherwise return code is 0
+    retcode2 = call(['git', 'diff', '--cached', '--exit-code'])
```

Comment:
It was, but only because I noticed the second command did not produce stdout on my terminal.  I've redirected it now though in the event that the command changes or produces stdout on Apple unbeknownst to me.